data/file_level_aggregation.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import pandas as pd
+
+
+class FileData:
+    def __init__(self) -> None:
+        self.file_name = None
+        self.repo_name = None
+        self.data = defaultdict(list)
+
+    def __repr__(self) -> str:
+        return str(self.data)
+
+
+class DataAggregator:
+    def __init__(self, file: str) -> None:
+        self._data = pd.read_csv(file)
+
+    def aggregate(self) -> dict:
+        print("Aggregating pull request data to file level")
+        file_level_data = defaultdict(FileData)
+        for idx in range(len(self._data)):
+            repo_name = self._data.iloc[idx]['repo name']
+            author = self._data.iloc[idx]['author']
+            pull_request_id = int(self._data.iloc[idx]['pull request id'])
+            pull_request_created_time = float(
+                self._data.iloc[idx]['pull request created time'])
+            pull_request_closed_time = float(
+                self._data.iloc[idx]['pull request closed time'])
+            pull_request_review_time = float(
+                self._data.iloc[idx]['pull request review time'])
+            reverted_pull_request_id = int(
+                self._data.iloc[idx]['reverted pull request id'])
+            pull_request_revert_time = float(
+                self._data.iloc[idx]['pull request revert time'])
+            num_review_comments = \
+                int(self._data.iloc[idx]['num review comments'])
+            review_comments_msg = \
+                eval(self._data.iloc[idx]['review comments msg'])
+            num_issue_comments = int(self._data.iloc[idx]['num issue comments'])
+            issue_comments_msg = \
+                eval(self._data.iloc[idx]['issue comments msg'])
+            num_approved_reviewers = \
+                int(self._data.iloc[idx]['num approved reviewers'])
+            approved_reviewers = \
+                eval(self._data.iloc[idx]['approved reviewers'])
+            num_commits = int(self._data.iloc[idx]['num commits'])
+            num_line_changes = int(self._data.iloc[idx]['num line changes'])
+            file_changes = eval(self._data.iloc[idx]['files changes'])
+            file_versions = eval(self._data.iloc[idx]['file versions'])
+            check_run_results = eval(self._data.iloc[idx]['check run results'])
```

Comment:
To avoid code redundancy  and improve readability:

1. Maintain a dict of column name to function:
columns = {'created time': int, 'issue comments msg': eval .........}

2. Iterate over this dict and store the values in a dict
values[key] = columns[key](self._data.iloc[idx][key])

3. Use this values dict for file_level_data

Note: Use the same key name (or same pattern) for both the csv file and file_level_data.